### ISSUE TITLE
Split memory and line profiler into separate plugins?

### ISSUE THREAD

**Issue by goanpeca** [spyder-ide/spyder.line_profiler#2]
Hi @Nodd do you think it makes sense to split this plugin into two?

Otherwise I can adjust the code to allow for instance doing this.

``` python
# =============================================================================
# The following statements are required to register this 3rd party plugin:
# =============================================================================
from .lineprofiler import LineProfiler
from .memoryprofiler import MemoryProfiler

PLUGIN_CLASS = [LineProfiler, MemoryProfiler]
```

Let me know what you think :-)

**Comment by Nodd** [comment]
It should be split. I wasn't sure at first, then I received a contribution to implement it and I merged it. But even if they work similarly, they are different tool that test completely different things and are displayed in a diferent panel.
Also the code for the memory profiler should be refactored, right now the data structure is reorganised twice, one th put it in the same shape as line_profiler, and a second one to put it in the table. (again, I merged the contribution as is).
